.github/workflows/ci.yml

```diff
@@ -178,6 +178,162 @@ jobs:
+
+      - name: Check out code into the Go module directory
+        uses: percona-platform/checkout@v2
+
+      - name: Enable Go modules cache
+        uses: percona-platform/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ matrix.os }}-go-${{ matrix.go-version }}-modules-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-modules-
+
+      - name: Enable Go build cache
+        uses: percona-platform/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ matrix.os }}-go-${{ matrix.go-version }}-build-${{ github.ref }}-${{ hashFiles('**') }}
+          restore-keys: |
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-build-${{ github.ref }}-
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-build-
+
+      - name: Download Go modules
+        run: go mod download
+
+      - name: Install tools
+        run: |
+          make init
+          mkdir -p "${DBAAS_TOOLS_PATH}"
+          echo "${DBAAS_TOOLS_PATH}" >> $GITHUB_PATH          
+          curl -o aws-iam-authenticator https://amazon-eks.s3.us-west-2.amazonaws.com/1.19.6/2021-01-05/bin/linux/amd64/aws-iam-authenticator
+          chmod +x ./aws-iam-authenticator
+          mv aws-iam-authenticator "${DBAAS_TOOLS_PATH}"
+          # dbaas-controller tries to search for kubectl at certain path
+          cp $(which kubectl) "${DBAAS_TOOLS_PATH}/kubectl-1.16"
+        env:
+          DBAAS_TOOLS_PATH: "/opt/dbaas-tools/bin"
+
+      - name: Initialize CI environment
+        run: make ci-init
+
+      - name: Generate code
+        run: make gen
+
+      - name: Install binaries
+        run: make release install
+
+      - name: Check that API tests can be built
+        run: make test-api-build
+
+      - name: Setup environment for tests to run
+        run: |
+           export PATH_TO_KUBECONFIG="${HOME}/.kube/config"
+           mkdir -p "${PATH_TO_KUBECONFIG%/*}"
+           echo "${BASE64_KUBECONFIG}" > "${PATH_TO_KUBECONFIG}"
+           export RANDOM_STRING=$(head -500 /dev/urandom | tr -dc 'a-z0-9' | fold -w 16 | tail -n +1 | head -n 1)
+           export NAMESPACE="dbaas-controller-test-${RANDOM_STRING}"
+           make eks-setup-test-namespace NAMESPACE="${NAMESPACE}"
+        env:
+           BASE64_KUBECONFIG: ${{secrets.BASE64_KUBECONFIG}}
+
+      - name: Deploy operators to the test namespace
+        run: |
+          make eks-install-operators
+
+      - name: Run tests
+        run: |
+          go clean -testcache
+          make test-crosscover
+
+      - name: Upload coverage results
+        uses: percona-platform/codecov-action@v1
+        with:
+          file: crosscover.out
+          flags: crosscover
+          env_vars: GO_VERSION,KUBERNETES_VERSION
+          fail_ci_if_error: false
+      
+      - name: Delete the test namespace
+        if: ${{ always() }}
+        run: |
+          make eks-delete-current-namespace
```

Comment:
In my experience (not much) if we don't delete the operators there are garbage left.
For example, network interfaces, cloud formations, etc because those resources are owned by the operator so only deleting the namespace won't be enough.
Can we test that and add a line to uninstall operators first?

Reply:
Let me investigate this. I thought when one deletes namespace all its resources die with it.

Reply:
Did not find any issues with network interfaces not being deleted but proper clean up won't do any harm. I will delete operators.

Reply:
Deleting of operator deployments added, also secrets are being deleted now. Can't delete CRD because they are cluster scoped, it would break other tests running in parallel.